src/Take.Blip.Client/Extensions/Tunnel/EnvelopeExtensions.cs

```diff
@@ -7,8 +7,16 @@ public static class EnvelopeExtensions
     {
         public static bool TryGetTunnelFromEnvelope<T>(this T envelope, out Takenet.Iris.Messaging.Resources.Tunnel tunnelInformation) where T : Envelope
         {
-            if (envelope.From?.Domain != null &&
-                envelope.From.Domain.Equals(TunnelExtension.TunnelAddress.Domain, StringComparison.OrdinalIgnoreCase) &&
+            Node fromNode = envelope.From;
+
+            if (envelope.Metadata != null && envelope.Metadata.ContainsKey(Constants.ORIGINAL_SUBFLOW_REDIRECT_FROM))
```

Comment:
This method is called on the GetUserOwnerIdentitiesAsync in UserOwnerResolver, maybe its better to add a if there, instead of change the tunnelFromEnvelope if possible

Reply:
I do this so as not change the "envelope.from" in the "GetUserOwnerIdentitiesAsync"...

_tunnelExtension.TryGetTunnelAsync receives only "message" parameter